viostor/virtio_stor_hw_helper.c

```diff
@@ -43,6 +43,15 @@
                       pa = va ? StorPortGetPhysicalAddress(DeviceExtension, NULL, va, &len).QuadPart : 0; \
                     }
 
+#define MESSAGENUMBER_TO_QUEUE() { \
+                    MessageId = param.MessageNumber; \
+                    QueueNumber = MessageId - 1; \
+                    if (QueueNumber >= adaptExt->num_queues) { \
+                        QueueNumber %= adaptExt->num_queues; \
+                        MessageId += 1; \
```

Comment:
Why would this happen if the number of queues is limited during init?

Reply:
It is a kind of sanity check. I was not able create such situation yet, but if you hotplug a vCPU and crate an application which creates a thread, set the thread affinity to this new CPU and then run a disk IO operation , then
CPU index (queue number) in this case will be bigger than actual number of queues. Unless you don't specify  some more num_queues in the qemu command line, of cause. 

Cheers,
Vadim.